### DISCUSSION TITLE
Inconsistent geometry reporting following Trim operation in perl

### DISCUSSION THREAD

**Discussion by MikeTough** [ImageMagick/ImageMagick#5876] [Help]
Hi,

Version: ImageMagick 7.1.0-51 Q16-HDRI x86_64 20499

I've been using the Trim operation in perl and have been trying to get the trimmed width, height and offset parameter for further processing.

My original image is 4712x6874, and the trimmed image is 4510x6722+95+69.

Identify reports the following:
Geometry: 4510x6722+0+0 -- Trimmed image WxH, but original image offset
Base geometry: 4712x6874 -- Correct original image/canvas size
Page geometry: 4712x6874+95+69 -- Original image dimensions with trimmed offset (should this be trimmed WxH ?)
Origin geometry: +95+69 -- Correct offset for trimmed image

In perl
Get('page') returns 4712x6874+95+69 -- Original image WxH with trimmed offset (undocumented option)
Get('geometry') fails !!! (this option is documented, so should function)
Get('width') returns 4510 - trimmed width
Get('height') returns 6722 - trimmed width
Get('page.x) returns 95 - trimmed offset
Get('page.y) returns 69 - trimmed offset 

Clearly I can get what I need (the trimmed dimensions) from Get('width', 'height', 'page.x', 'page.y').

The documented Get('geometry') fails for me. Is this a bug?
The reporting for Geometry and Page Geometry appear confused. Each with part Original part Trimmed information. Is this a bug?

**Comment by snibgo** [comment]
I can't comment on the perl aspects. Perhaps "Get('geometry') fails !!!" is an IM bug.

> (this option is documented, so should function)

Please give the documentation URL.

Trimming without the equivalent of "+repage" will result in an output image that sits on a "page", aka "canvas". The output image width and height is generally smaller than the input. The page width and height are the input width and height, and page offsets (page.x and page.y) are the top-left coordinates of the output image on the input page. So `Get('page')` gives what I would expect.

**Reply by MikeTough** [reply]
URL for the perl documentation is : https://imagemagick.org/script/perl-magick.php
The relevant section is under the heading "Get an Image Attribute" and 'geometry' is listed in the scroll box of attributes. (Note 'page' is not listed there, but works for me).

With regard to the coordinates, you can see that the Get('page') and Identify "Page Geometry" does indeed give the correct offsets for the trimmed image, however the width and height given in Get('page') and Identify Geometry gives the untrimmed dimension resulting in a meaningless geometry. It returns 4712x6874+95+69 rather than 4510x6722+95+69 (which would correctly define the trimmed image)

Similarly, the Identify "Geometry" also returns a meaningless geometry of 4510x6722+0+0 rather than 4712x6874+0+0 (which would correctly define the original image.

I guess I'm saying that I would have expected the Identify "Geometry" to represent the full/original/ image/canvas and for Identify "Page Geometry" to represent the trimmed image within the original image/canvas, whereas it looks like neither returns a meaningful geometry.

I'm not a regular user of IM, so I may be being a bit stupid or have misunderstood the purpose/meaning of the geometry format. If so, please say so :) :) :)

**Comment by snibgo** [comment]
The current semantics of "page" (width and height of canvas, and offset into that canvas) are used extensively within IM, and is exposed in APIs. I doubt that it will change.
